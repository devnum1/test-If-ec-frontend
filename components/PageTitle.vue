// 01EZTX9M3AWK0ZB16B6KYR56F1

<template>
  <div class="page-title">
    <div class="page-title__wrapper container">
      <div v-if="btnBack" class="page-title__back">
        <NuxtLink class="page-title__link" :to="btnBack">
          <i class="icon-fonts icon-fonts--back" />
        </NuxtLink>
      </div>
      <div v-if="logo" class="text-center header-logo">
        <img src="~/assets/images/logo/1line_white.svg" />
      </div>
      <div v-else class="page-title__text">
        {{ title }}
      </div>
    </div>
  </div>
</template>

<script>
export default {
  props: {
    title: {
      type: String,
      default: '',
    },
    btnBack: {
      type: String,
      default: null,
    },
    logo: {
      type: Boolean,
      default: false,
    },
  },
};
</script>

<style lang="scss">
@import '~/assets/scss/components/page-title';
</style>
