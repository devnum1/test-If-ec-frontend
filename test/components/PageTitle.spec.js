import { shallowMount, config } from '@vue/test-utils';
import PageTitle from '@/components/PageTitle.vue';

config.mocks.$t = (key) => key;

describe('PageTitle', () => {
  let wrapper;

  const factory = () => {
    return shallowMount(PageTitle, {
      propsData: {
        title: 'Test title',
      },
    });
  };

  beforeEach(() => {
    wrapper = factory();
  });

  afterEach(() => {
    wrapper.destroy();
  });

  test('is a Vue instance', () => {
    const wrapper = shallowMount(PageTitle);
    expect(wrapper.vm).toBeTruthy();
  });

  test('match Snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('match the title', () => {
    expect(wrapper.find('.page-title__text').text()).toEqual('Test title');
  });
});
