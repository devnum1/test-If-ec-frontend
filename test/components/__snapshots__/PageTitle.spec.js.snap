// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTitle match Snapshot 1`] = `
"<div class=\\"page-title\\">
  <div class=\\"page-title__wrapper container\\">
    <!---->
    <div class=\\"page-title__text\\">
      Test title
    </div>
  </div>
</div>"
`;
